gateway/src/clickhouse_migration_manager/migrations/migration_0003.rs

```diff
@@ -94,100 +94,169 @@ impl<'a> Migration for Migration0003<'a> {
-                             ADD COLUMN tags Map(String, String) DEFAULT map();"#;
+        let query = r#"
+            ALTER TABLE DemonstrationFeedback
+            ADD COLUMN IF NOT EXISTS tags Map(String, String) DEFAULT map();"#;
         let _ = self.clickhouse.run_query(query.to_string()).await?;
 
         // Add a column `tags` to the `FloatMetricFeedback` table
-        let query = r#"ALTER TABLE FloatMetricFeedback
-                             ADD COLUMN tags Map(String, String) DEFAULT map();"#;
+        let query = r#"
+            ALTER TABLE FloatMetricFeedback
+            ADD COLUMN IF NOT EXISTS tags Map(String, String) DEFAULT map();"#;
         let _ = self.clickhouse.run_query(query.to_string()).await?;
 
-        // In the following few queries we create the materialized views that map the tags from the original tables to the new `FeedbackTags` table
+        // In the following few queries we create the materialized views that map the tags from the original tables to the new `FeedbackTag` table
         // We do not need to handle the case where there are already tags in the table since we created those columns just now.
         // So, we don't worry about timestamps for cutting over to the materialized views.
-        // Create the materialized view for the `FeedbackTags` table from BooleanMetricFeedback
+        // Create the materialized view for the `FeedbackTag` table from BooleanMetricFeedback
         let query = r#"
-            CREATE MATERIALIZED VIEW IF NOT EXISTS BooleanMetricFeedbackTagsView
-            TO FeedbackTags
+            CREATE MATERIALIZED VIEW IF NOT EXISTS BooleanMetricFeedbackTagView
+            TO FeedbackTag
             AS
                 SELECT
                     metric_name,
-                    entry.1 as tag_key,
-                    entry.2 as tag_value,
+                    entry.1 as key,
+                    entry.2 as value,
                     id as feedback_id
                 FROM BooleanMetricFeedback
                 ARRAY JOIN arrayZip(mapKeys(tags), mapValues(tags)) AS entry
             "#;
         let _ = self.clickhouse.run_query(query.to_string()).await?;
 
-        // Create the materialized view for the `FeedbackTags` table from CommentFeedback
+        // Create the materialized view for the `FeedbackTag` table from CommentFeedback
         let query = r#"
-            CREATE MATERIALIZED VIEW IF NOT EXISTS CommentFeedbackTagsView
-            TO FeedbackTags
+            CREATE MATERIALIZED VIEW IF NOT EXISTS CommentFeedbackTagView
+            TO FeedbackTag
             AS
                 SELECT
                     'comment' as metric_name,
-                    entry.1 as tag_key,
-                    entry.2 as tag_value,
+                    entry.1 as key,
+                    entry.2 as value,
                     id as feedback_id
                 FROM CommentFeedback
                 ARRAY JOIN arrayZip(mapKeys(tags), mapValues(tags)) AS entry
```

Comment:
Are you 100% sure this works? Where is this snippet from? I couldn't find in CH docs.

Failure case: `mapKeys` and `mapValues` don't use the same order for some reason.

Why not retrieve all the keys, then `k as key`, `map[k] as value` or similar?

Reply:
it's from iteratively playing with ClickHouse -- I can change it to this as a more idiot-proof solution but I am entirely confident this does the job. WIll fix.